app/src/main/java/com/cornellappdev/uplift/networking/ApolloAdapters.kt

```diff
@@ -1,265 +1,182 @@
-                hours = courtHours[day]?.timeRanges?.get(courtIndex)?.let { range ->
-                    listOf(
-                        TimeInterval(
-                            parseTimeOfDay(range.startTime.toString()),
-                            parseTimeOfDay(range.endTime.toString())
-                        )
-                    )
-                } ?: listOf()
-            )
-        }
+    if (highCap.isNaN()) return null
 
-        return list
-    }
+    // Ex: "2023-09-19T18:42:00"
+    val cal = Calendar.getInstance()
+    val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.ENGLISH)
+    cal.time = sdf.parse(facility.capacity.updated.toString()) ?: Calendar.getInstance().time
+
+    return UpliftCapacity(
+        percent = facility.capacity.percent,
+        updated = cal
+    )
+}
 
-    return List(7) { day ->
-        GymnasiumInfo(gymnasiumHours[day], getCourtInfosForDay(day))
+/**
+ * Returns the "FITNESS" facility at a gym. `null` if the gym does not have it.
+ */
+private fun GymListQuery.Gym.getFitnessFacility(): GymListQuery.Facility? {
+    return facilities?.find { facility ->
+        facility?.facilityType.toString() == "FITNESS"
     }
 }
 
-fun GymListQuery.Gym.toUpliftGym(): UpliftGym {
-    return UpliftGym(
-        name = name,
-        id = id,
-        popularTimes = parsePopularTimes(popular),
-        imageUrl = imageUrl ?: defaultGymUrl,
-        hours = parseHours(times),
-        equipmentGroupings = pullEquipmentGroupings(this),
-        miscellaneous = pullMiscellaneous(this),
-        bowlingInfo = pullBowling(this),
-        // No swimming info in backend yet...
-        swimmingInfo = null,
-        gymnasiumInfo = pullGymnasiumInfos(this),
-        // TODO: Reflect actual capacity pulled from backend.
-        capacity = Capacity(Pair((Math.random() * 20 + 100).toInt(), 140), Calendar.getInstance())
-    )
+/**
+ * Returns a list of all the [UpliftGym]s that this gym query represents.
+ *
+ * Example: Teagle Gym in backend has both Teagle Up and Teagle Down as separate fitness centers.
+ * This should separate them into distinct gyms.
+ */
+fun GymListQuery.Gym.toUpliftGyms(): List<UpliftGym> {
+    val fitnessFacilities = facilities?.filter { facility ->
+        facility?.facilityType.toString() == "FITNESS"
+    } ?: listOf()
+
+    return fitnessFacilities.filterNotNull().map { facility ->
+        UpliftGym(
+            name = facility.name,
+            id = facility.id,
+            popularTimes = pullPopularTimes(facility),
+            // Need replace because there's a typo with the single quote.
+            imageUrl = imageUrl?.replace("'", "") ?: defaultGymUrl,
+            hours = pullHours(facility),
+            equipmentGroupings = pullEquipmentGroupings(facility),
+            miscellaneous = pullMiscellaneous(),
+            bowlingInfo = pullBowling(),
+            // No swimming info in backend yet...
+            swimmingInfo = pullSwimmingInfo(),
```

Comment:
do you know when swimming will be added lol

Reply:
Oh that was an old comment when we had the old backend. In fact a LOT of stuff isn't in backend yet lol hence the like million placeholder functions. I don't think swimming info's coming for a while.